packages/kit/src/nitro.ts

```diff
@@ -99,15 +99,14 @@ export function addServerImports (imports: Import[]) {
 }
 
 /**
- * Add directories to be scanned by Nitro
+ * Add directories to be scanned for auto-imports by Nitro
  */
 export function addServerImportsDir (dirs: string | string[], opts: { prepend?: boolean } = {}) {
   const nuxt = useNuxt()
+  const _dirs = Array.isArray(dirs) ? dirs : [dirs]
   nuxt.hook('nitro:config', (config) => {
-    config.scanDirs = config.scanDirs || []
-
-    for (const dir of (Array.isArray(dirs) ? dirs : [dirs])) {
-      config.scanDirs[opts.prepend ? 'unshift' : 'push'](dir)
-    }
+    config.imports = config.imports || {}
```

Comment:
```suggestion
    config.imports ??= config.imports
```

Reply:
nice - but I think we want to do this in parallel with above implementation (L92-97)